### PR TITLE
fix(workspaces): workspace design review

### PR DESCRIPTION
**Pull request by foochifa**
## Problem

Design review with @staceytan1998, problems raised:
1. Change 'All forms' style to `subhead-3` and it's form count to `subhead-2`
2. Fixed width for workspace desktop menu in `RowActionsDropdown`, and truncate if workspace name is too long
3. Bold folder name in delete modal
4. Separate the name and number from `WorkspaceHeader` and truncate the workspace name if it is too long
5. Use drawer for mobile view of `WorkspaceEditMenu`
6. Name change from workspace -> folder
7. Different `EmptyWorkspace` messages based on whether a folder is selected or not, [design here](https://www.figma.com/file/CMq0CJtawn7NGM7Zdpi8Ti/Form-Design-Master?type=design&node-id=4139-73642&mode=design&t=deVHpt5rCPTrCyab-4)! 

## Solution

**Breaking Changes** 

- [ ] No - this PR is backwards compatible  

## Before & After Screenshots

**BEFORE**:

**AFTER**:

### CODE REVIEW

**Comment by foochifa** [comment]
@justynoh I realised this PR modifies and builds on a lot of what you did for #5458, maybe you would like to take a look too? Thank you! 🙏

File: frontend/src/features/workspace/components/WorkspaceModals/RenameWorkspaceModal.tsx [hunk 1]
```diff
@@ -76,10 +76,10 @@ export const RenameWorkspaceModal = ({
     <Modal isOpen={isOpen} onClose={onClose} size={modalSize}>
       <ModalOverlay />
       <ModalContent>
-        <ModalHeader>Rename workspace</ModalHeader>
+        <ModalHeader>Rename Folder</ModalHeader>
```
Comment (justynoh) [inline frontend/src/features/workspace/components/WorkspaceModals/RenameWorkspaceModal.tsx]: supernit: small f

File: frontend/src/features/workspace/components/WorkspaceHeader/WorkspaceEditMenu.tsx [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+}): JSX.Element => {
+  return (
+    <Menu placement="bottom-start">
+      {({ isOpen }) => (
+        <>
+          <MenuButton
+            as={IconButton}
+            _active={{ bg: 'secondary.100' }}
+            isActive={isOpen}
+            aria-label="Edit Workspace"
```
Comment (justynoh) [inline frontend/src/features/workspace/components/WorkspaceHeader/WorkspaceEditMenu.tsx]: aria label should be folder right?

File: frontend/src/features/workspace/mutations.ts [hunk 3]
```diff
@@ -197,7 +197,7 @@ export const useWorkspaceMutations = () => {
   const createWorkspaceMutation = useMutation(
     (params: { title: string }) => createWorkspace(params),
     {
-      onSuccess: () => handleSuccess('New Workspace Created.'),
+      onSuccess: () => handleSuccess('New Folder Created.'),
```
Comment (justynoh) [inline frontend/src/features/workspace/mutations.ts]: Checking that all the `Folder`s in this file should be capital F or small F? also this particular toast should be sentence case right, not title case.

**Review by justynoh — APPROVED** [review]
Code wise LGTM, just some supernits about copy

File: frontend/src/features/workspace/components/EmptyWorkspace/EmptyWorkspace.tsx [hunk 4]
```diff
@@ -24,28 +26,31 @@ export const EmptyWorkspace = ({
...
       </Text>
-      <Text
-        textStyle="body-1"
-        color="secondary.500"
-        mb={{ base: '2.5rem', md: '2rem' }}
-      >
-        Get started by creating a new form
+      <Text textStyle="body-1" color="secondary.500">
+        {isFolder
+          ? 'Organise your forms by grouping them in to folders'
```
Comment (wanlingt) [inline frontend/src/features/workspace/components/EmptyWorkspace/EmptyWorkspace.tsx]: nit: `into` instead of `in to`?

**Review by wanlingt — APPROVED** [review]
Folders look great!

Should we also rename all the Workspace__ components to Folder__?

Also some questions which are not in this PR's scope:
- How do I remove a form from a folder? Currently the menu only shows 'add to folder'
- Will there be a select multiple forms option? I imagine that people usually move items to folders in bulk (at least I do hahaha)

**Comment by foochifa** [comment]
Thanks @justynoh  @wanlingt for the review! Copy changes here [0354749](https://github.com/opengovsg/FormSG/pull/6632/commits/035474936c701d9069f12bab1e5699b5348e68a6)

> How do I remove a form from a folder? Currently the menu only shows 'add to folder'

Ahhh yes currently there isn't really a way to do that 😅 will discuss with @staceytan1998 on this! Thanks for highlighting it @wanlingt 

> Will there be a select multiple forms option? I imagine that people usually move items to folders in bulk (at least I do hahaha)

Yep! in the backlog!

**Pull request merged**